### PR TITLE
Dev command for MESG Framework

### LINKED ISSUE #189: MESG Framework
Improve the CLI for local development
- [x] Simplify the management on the local engine (https://github.com/mesg-foundation/js-sdk/pull/188)
  - [x] start and stop engine automatically in `dev` commands (https://github.com/mesg-foundation/js-sdk/pull/188)
  - [x] inject account in the engine (https://github.com/mesg-foundation/js-sdk/pull/188)
- [x] Remove confusing commands (list, detail, create, compile, delete, daemon) (https://github.com/mesg-foundation/js-sdk/pull/188)
- [x] MESG framework (https://github.com/mesg-foundation/js-sdk/pull/198)
- [ ] update docs.mesg.com once done (https://github.com/mesg-foundation/docs/pull/121)

### PR DESCRIPTION
**Pull request by antho1404**
A proposition to normalize all the development of apps in mesg.
The idea is to have a command `dev` that will automatically deploy your processes/services based on a specific file organization.

Here is an idea of an organization.

```bash
├── .env
├── processA.yml
├── processB.yml
├── processX.yml
├── services
│   ├── service-name-1
│   │   ├── mesg.yml
│   │   ├── ...other files
│   ├── service-name-2
│   │   ├── mesg.yml
│   │   ├── ...other files
```

A command `dev` would read the `services` directory and compile/deploy each service. It will also deploy all the `.yml` in the root of the project and inject the environment variable contained in the `.env`.

## Note for later

With this in mind, we can also have a "service importer", the same way that you can import a library with npm with `npm install xxx` we can have a command `link/add/install` that will download a service and put it in the `services` directory.

We might add these services in a gitignore or instead copy them in a `node_module/vendor` like a directory that users will choose to version or no while still able to have their own local services in the `services` directory.

### CODE REVIEW

**Comment by NicolasMahe** [comment]
I really like this concept. I don't have much to add for now ;)

**Comment by blackinno** [comment]
I agree with this concept and also it would be nice if we can have a command like `mesg-cli service:import <NAME of Service>` as @antho1404 said because it easier than to find services in GitHub repo.

File: packages/cli/src/commands/dev.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  private grpc = new API('localhost:50052')
+  private logs: GRPCStream<IExecution>
+  private services: IService[] = []
+  private processes: IProcess[] = []
+  private runners: IRunner[] = []
+
+  async run() {
+    const { args, flags } = this.parse(Dev)
+
+    if (!existsSync(join(args.PATH, 'services'))) this.error(`${args.PATH} is not a recognized as a MESG project`)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/dev.ts:51]: why forcing to have a services folder?
what about application that use only services from github?

**Review by NicolasMahe — COMMENTED** [review]
Got the following error after pressing ctrl+c.
```
  ✔ Starting environment
  ✔ Creating process "process-test.yml"
  ✔ Fetching logs
[test][node-1] - E6Hh1tKL3pi1bi7iL3FH6y1Um1vHfCoo8qZfGVFfm9pp - execute
	inputs:  {"code":"module.export = function(value) { return true }","inputs":{"foo":42}}
	outputs: {"result":true}

[test][node-1] - E6Hh1tKL3pi1bi7iL3FH6y1Um1vHfCoo8qZfGVFfm9pp - execute
	inputs:  {"inputs":{"foo":42},"code":"module.export = function(value) { return true }"}
	outputs: {"result":true}

[test][node-1] - E6Hh1tKL3pi1bi7iL3FH6y1Um1vHfCoo8qZfGVFfm9pp - execute
	inputs:  {"code":"module.export = function(value) { return true }","inputs":{"foo":42}}
	outputs: {"result":true}

[test][node-1] - E6Hh1tKL3pi1bi7iL3FH6y1Um1vHfCoo8qZfGVFfm9pp - execute
	inputs:  {"code":"module.export = function(value) { return true }","inputs":{"foo":42}}
	outputs: {"result":true}

^C ›   Warning: Result stream error: 1 CANCELLED: Cancelled
  ✔ Stopping logs
  ✖ Stopping running services
    → unauthorized: signature verification failed; verify correct account sequence and chain-id
    Deleting processes
    Stopping environment
(node:11450) UnhandledPromiseRejectionWarning: Error: unauthorized: signature verification failed; verify correct account sequence and chain-id
    at API.<anonymous> (/Users/nico/Development/MESG/js-sdk/packages/api/lib/lcd.js:58:23)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/nico/Development/MESG/js-sdk/packages/api/lib/lcd.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:11450) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:11450) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Review by NicolasMahe — COMMENTED** [review]
a command `init` could be nice to init a folder with a `.env`, one basic process and one basic service

File: packages/cli/src/commands/dev.ts (hunk 1 above)
Comment (antho1404) [reply packages/cli/src/commands/dev.ts:51]: initially, the idea was to have the `link` command that will populate the `service` directory with the one from github or any to have everything in local.
The goal is also to always have the same structure even if it's empty (can put a `.gitkeep`) just to keep it as a framework where you don't necessarily use all the features

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**